mathjax3-ts/output/svg/Wrapper.ts

```diff
@@ -0,0 +1,424 @@
+            transform = 'translate(' + this.fixed(x) + ', ' + this.fixed(y) + ')'
+                      + (transform ? ' ' + transform : '');
+            this.adaptor.setAttribute(element, 'transform', transform);
+        }
+    }
+
+    /**
+     * @param {number} n        The character number
+     * @param {number} x        The x-position of the character
+     * @param {number} y        The y-position of the character
+     * @param {N}               The container for the character
+     * @param {string} variant  The variant to use for the character
+     * @return {number}         The width of the character
+     */
+    public placeChar(n: number, x: number, y: number, parent: N, variant: string = null) {
+        if (variant === null) {
+            variant = this.variant;
+        }
+        const C = n.toString(16).toUpperCase();
+        const [h, d, w, data] = this.getVariantChar(variant, n);
+        if ('p' in data) {
+            this.place(x, y, this.adaptor.append(parent, this.svg('path', {
+                'data-c': C, d: (data.p ? 'M' + data.p + 'Z' : '')
+            })));
+        } else if ('c' in data) {
+            const g = this.adaptor.append(parent, this.svg('g', {'data-c': C}));
+            this.place(x, y, g);
+            x = 0;
+            for (const n of this.unicodeChars(data.c)) {
+                x += this.placeChar(n, x, y, g, variant);
+            }
+        } else if (data.unknown) {
+            const char = String.fromCharCode(n);
+            const text = this.adaptor.append(parent, this.jax.unknownText(char, variant));
+            this.place(x, y, text);
+            return this.jax.measureTextNodeWithCache(text, char, variant).w;
+        }
+        return w;
+    }
+
+    /**
+     * @param {number} W       The total width
+     * @param {number} w       The width to be aligned
+     * @param {string} align   How to align (left, center, right)
+     * @return {number}        The x position of the aligned width
+     */
+    protected getAlignX(W: number, w: number, align: string) {
+        if (align === 'right') return W - w ;
+        if (align === 'left') return 0;
+        return (W - w) / 2;
+    }
+
+    /**
+     * @param {number} H        The total height
+     * @param {number} D        The total depth
+     * @param {number} h        The height to be aligned
+     * @param {number} d        The depth to be aligned
+     * @param {string} align    How to align (top, bottom, middle, axis, baseline)
+     * @return {number}         The y position of the aligned baseline
+     */
+    protected getAlignY(H: number, D: number, h: number, d: number, align: string) {
+        if (align === 'top') return H - h ;
+        if (align === 'bottom') return d - D;
+        if (align === 'middle') return ((H - h) - (D - d)) / 2;
+        return 0; // baseline and axis
+    }
+
+    /*******************************************************************/
+    /**
+     * For debugging
```

Comment:
Maybe we should have a naming convention for methods that should eventually be removed.